src/app/mytools/mytools.service.ts

```diff
@@ -54,6 +58,78 @@ export class MytoolsService extends MyEntriesService {
         }
       );
   }
-
+  selectEntry(tool: DockstoreTool | Workflow | null): void {
+    if (tool !== null) {
+      this.containersService.getContainer(tool.id, includesValidation).subscribe(result => {
+        this.location.go('/my-tools/' + result.tool_path);
+        this.containerService.setTool(result);
+      });
+    }
+  }
   registerEntry(entryType: EntryType) {}
+
+  convertToolsToOrgToolObject(tools: DockstoreTool[], selectedTool: DockstoreTool): OrgToolObject[] {
+    if (!tools) {
+      return [];
+    }
+    const orgToolObjects: OrgToolObject[] = [];
+    tools.forEach(tool => {
+      const existingOrgToolObject = orgToolObjects.find(
+        orgToolObject => orgToolObject.registry === tool.registry_string && orgToolObject.namespace === tool.namespace
+      );
+      if (existingOrgToolObject) {
+        if (tool.is_published) {
+          existingOrgToolObject.published.push(tool);
+        } else {
+          existingOrgToolObject.unpublished.push(tool);
+        }
+      } else {
+        const newOrgToolObject: OrgToolObject = {
+          registry: tool.registry_string,
+          namespace: tool.namespace,
+          ...this.createOrgEntriesObject(tool)
+        };
+        orgToolObjects.push(newOrgToolObject);
+      }
+    });
+    this.recursiveSortOrgToolObjects(orgToolObjects);
+    this.setExpand(orgToolObjects, selectedTool);
+    return orgToolObjects;
+  }
+
+  protected recursiveSortOrgToolObjects(orgToolObjects: OrgToolObject[]) {
+    this.sortEntriesOfOrgEntryObjects(orgToolObjects);
+    orgToolObjects.sort(this.sortOrgToolObjects);
+  }
+
+  protected sortOrgToolObjects(orgToolObjectA: OrgToolObject, orgToolObjectB: OrgToolObject): number {
+    const keyA = [orgToolObjectA.registry, orgToolObjectA.namespace].join('/').toLowerCase();
+    const keyB = [orgToolObjectB.registry, orgToolObjectB.namespace].join('/').toLowerCase();
+    return keyA.localeCompare(keyB);
+  }
+
+  /**
+   * Set the expanded property on the OrgToolObjects
+   *
+   * @protected
+   * @param {OrgToolObject[]} orgToolObjects
+   * @param {DockstoreTool} selectedTool
+   * @returns
+   * @memberof MytoolsService
+   */
+  protected setExpand(orgToolObjects: OrgToolObject[], selectedTool: DockstoreTool) {
```

Comment:
`selectedTool: DockstoreTool | null` (since you're doing a null check on the next line).